packages/plugin-ext/src/hosted/node/hosted-plugin-process.ts

```diff
@@ -41,20 +42,41 @@ export class HostedPluginProcess implements ServerPluginRunner {
     @inject(HostedPluginCliContribution)
     protected readonly cli: HostedPluginCliContribution;
 
+    @inject(HostedPluginProcessesCache)
+    protected readonly pluginProcessCache: HostedPluginProcessesCache;
+
     @inject(ContributionProvider)
     @named(PluginHostEnvironmentVariable)
     protected readonly pluginHostEnvironmentVariables: ContributionProvider<PluginHostEnvironmentVariable>;
 
     private childProcess: cp.ChildProcess | undefined;
+
     private client: HostedPluginClient;
 
+    private async getClientId(): Promise<number> {
+        let clientId = await this.client.getClientId();
```

Comment:
if `clientId` is greater than `clientIdSeq` we have to reject it to support the case when a server is restarted, otherwise they can be several clients with the same id